File: scripts/hooks/pre_commit.py
import shlex
import subprocess

branch = str.strip(
    subprocess.check_output(
        shlex.split("git rev-parse --symbolic-full-name --abbrev-ref HEAD")).
    decode('utf-8'))

if branch == "dev":
	subprocess.check_call(shlex.split("yarn run lint-staged"))

File: scripts/hooks/pre-commit.ts
#!/usr/bin/env -S npx ts-node-transpile-only --script-mode

import { commandSync } from 'execa';

function precommit() {
	const branch = commandSync(
		'git rev-parse --symbolic-full-name --abbrev-ref HEAD'
	).stdout.toString();

	if (branch === 'dev') {
		commandSync('pnpm run lint-staged', {
			stdout: process.stdout,
		});
	}
}

if (require.main === module) {
	precommit();
}
